### PR TITLE
Go `1.22`

### PR DESCRIPTION
**Pull request by filiptronicek**
## Description

Bumps Go to its latest version.

### CODE REVIEW

**Comment by kylos101** [comment]
Build is failing with: `Error: cannot test chunk lang-go:1.22: failed to solve: process "/bin/sh -c curl -fsSL https://dl.google.com/go/go$GO_VERSION.linux-amd64.tar.gz | tar xzs" did not complete successfully: exit code: 2`

**Comment by filiptronicek** [comment]
@kylos101 you're right, I came from the future, a world where a version such as `1.22.1` exists. Fixed that misdemeanour.

(accidentally looked at the [wrong page](https://dev.golang.org/release#Go1.22.1) for this)

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 A curated, chronologically ordered list of notable changes in [Gitpod's default workspace images](https://hub.docker.com/u/gitpod).
 
+## 2024-02-11
+
+- Bump Go to `1.22.0`
```
Comment (kylos101) [inline CHANGELOG.md:7]: ❤️

**Review by kylos101 — APPROVED** [review]
👍 we'll later have to upgrade go and rebuild here: https://github.com/gitpod-io/gitpod/blob/6678329a328aae9703d9b22b792d19d6fb7fdb72/dev/image/Dockerfile#L5

**Pull request merged**